server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -457,7 +457,7 @@ static RestoreDiff getRestoreDiff(Subscription subscription,
         HashSet<RelationName> relationNamesForStateUpdate = new HashSet<>();
         ArrayList<TableOrPartition> toRestore = new ArrayList<>();
         Metadata subscriberMetadata = subscriberState.metadata();
-        for (var indexName : stateResponse.concreteIndices()) {
+        for (var indexName : stateResponse.restorableIndices()) {
             IndexParts indexParts = IndexName.decode(indexName);
```

Comment:
we keep indices in publication response (to address reported issue and don't them to be treated as dropped) but we restore only those that we actually can restore - without this `test_deleted_and_recreated_partition_is_also_deleted_and_restored_on_subscriber` fails